pkg/api/controllers/volume.go

```diff
@@ -69,16 +70,30 @@ func (v *VolumePortal) CreateVolume() {
 	if volume.ProfileId == "" {
 		log.Warning("Use default profile when user doesn't specify profile.")
 		prf, err = db.C.GetDefaultProfile(ctx)
+		if err != nil {
+			errMsg := fmt.Sprintf("get profile failed: %s", err.Error())
+			v.ErrorHandle(model.ErrorBadRequest, errMsg)
+			return
+		}
 		volume.ProfileId = prf.Id
 	} else {
 		prf, err = db.C.GetProfile(ctx, volume.ProfileId)
+		if err != nil {
+			errMsg := fmt.Sprintf("get profile failed: %s", err.Error())
+			v.ErrorHandle(model.ErrorBadRequest, errMsg)
+			return
+		}
 	}
-	if err != nil {
-		errMsg := fmt.Sprintf("get profile failed: %s", err.Error())
-		v.ErrorHandle(model.ErrorBadRequest, errMsg)
-		return
+	// To get pool details for Thin OpenSDS
+	var pools []*model.StoragePoolSpec
+	if CONF.OsdsApiServer.InstallType == "thin" {
```

Comment:
Make "thin" a constant.

Reply:
I don't think it's a viable way to use "thin" as a type for choosing a pool.  Another solution for your information is: The api server can go to dock server directly if the requests contains "poolId"

Reply:
> I don't think it's a viable way to use "thin" as a type for choosing a pool. Another solution for your information is: The api server can go to dock server directly if the requests contains "poolId"

@wisererik  I am using data of dock server only which is stored in DB to get details like Pool-ID which is required to create a volume. Currently I don't think request contains Pool details.

Reply:
Hi Gamma, if using thin as a type in the system, we also need to implement a small selector to choose a pool from Dock in api server.  if using poolId (or poolName, because pool name is unique under the same Dock), the api server will get dock directly from pool.

Reply:
Hi @wisererik I have added a pool selecter as you suggested, Thanks.